v0/destinations/fb_custom_audience/transform.js

```diff
@@ -187,114 +186,149 @@ function prepareDataField(userSchema, userUpdateList) {
   return data;
 }
 
-function preparePayload(userUpdateList, userSchema, bufferPayload) {
-  const payload = {};
-  Array.isArray(userSchema)
-    ? (payload.schema = userSchema)
-    : (payload.schema = [userSchema]);
-  payload.data = prepareDataField(userSchema, userUpdateList);
-  if (isDefinedAndNotNull(bufferPayload.is_raw)) {
-    if (typeof bufferPayload.is_raw === "boolean") {
-      payload.is_raw = bufferPayload.is_raw;
-    } else {
+function preparePayload(
+  userUpdateList,
+  userSchema,
+  paramsPayload,
+  isHashRequired
+) {
+  const prepareFinalPayload = paramsPayload;
+  const { data_source } = paramsPayload;
+  if (Array.isArray(userSchema)) {
+    prepareFinalPayload.schema = userSchema;
+  } else {
+    prepareFinalPayload.schema = [userSchema];
+  }
+
+  prepareFinalPayload.data = prepareDataField(
+    userSchema,
+    userUpdateList,
+    isHashRequired
+  );
+  if (isDefinedAndNotNull(prepareFinalPayload.is_raw)) {
+    if (typeof prepareFinalPayload.is_raw !== "boolean") {
       logger.debug("is_raw field is not boolean, so dropping it");
+      delete prepareFinalPayload.is_raw;
```

Comment:
Let's move this logic to the calling function instead of deleting the property here. In the calling function if this `condition`  is false we will not set `is_raw` in the payload

same for `data_source`